graphql_compiler/compiler/helpers.py

```diff
@@ -328,6 +328,11 @@ def get_location_name(self):
         mark_name = u'__'.join(self.query_path) + u'___' + six.text_type(self.visit_counter)
         return (mark_name, self.field)
 
+    def is_revisited_at(self, other_location):
```

Comment:
I think the name is deceptive. "revisited" implies time dependency which this is not checking

Reply:
Agreed

Reply:
I think in this specific use case, we care that the new location is actually a *re*-visit -- the `other_location.visit_counter` has to be strictly greater than `self.visit_counter`, and I'm adding that to the code now.

I don't think it has to be an either-or though -- I'm open to refactoring this code or adding another helper method if there are use cases for it.